lib/Conversion/TTIRToTTIRDecomposition/TTIRToTTIRDecomposition.cpp

```diff
@@ -248,9 +253,8 @@ struct ConvolutionToConv2dPattern
       SPATIAL_DIM_WIDTH,
   };
 
-  LogicalResult isConv2d(ttir::ConvolutionOp op) const {
-
-    // Conv2d will have 2 spatial dimensions
+  LogicalResult isValidConv(ttir::ConvolutionOp op,
+                            uint32_t numSpatialDims) const {
```

Comment:
This method is kind of broken now. The intent is to check whether it's a valid conv, but it only compares spatial dims cnt vs the provided `numSpatialDims`, so if someone pushes a 3d conv thru, it will deem it valid. It's a strange pattern to have to provide the intended number of spatial dims to the fn. In my eyes, it makes more sense for the fn to check if the conv is 1d/2d vs something else.

Btw with this change you've overshadowed the more global `constexpr static uint32_t numSpatialDims = 2;` which was intended to signal that only 2d convs were valid.

Reply:
I guess we should think in terms of support, not validity. `isSupportedConv` would better convey the intent of the function after above-mentioned changes.

Reply:
With regards to change change of structure outlined in the comment written below, I have changed both of the validity/support checks (for `numSpatialDims` and otherwise) to live in the base `ConvolutionDecompositionPattern` class, which both `ConvolutionToConv2dPattern` and `Legalize1DConvolutionPattern` inherit from.
Please go through the changes regarding comments one more time and let me know if you think I missed something. Thanks!